### ISSUE TITLE
Expand the size of elasticsearch disk managed by elastic operator in kubernetes

### ISSUE THREAD

**Issue by chenj100** [chenj100/archive#2]
I'm using elastic operator (aka eck, elasticsearch on cloud) https://github.com/elastic/cloud-on-k8s/releases/tag/1.8.0

The elasticsearch created by the operator is a statefulset. So obviously this should following the usual way of expanding volume in kubernetes. e.g.

1. ensure the storage class supports volume expansion and has [allow-volume-expansion](https://kubernetes.io/docs/concepts/storage/storage-classes/#allow-volume-expansion) set to true
2. edit the request storage size in the pvc resource

However, in certain environment (e.g. AKS), the disks don't support online expansion, the disks can only be resized when they are not attached to the worker nodes. So this means we need to delete the statefulset (pods), wait for the resize to succeed, and recreate the statefulset.

However, the elastic operator seems to keep creating the statefulset when I delete it, which doesn't give enough time for the disk to be dettached. When I delete the elasticsearch crd, the pvc are deleted as well. This isn't aligned with the default kubernetes behavior (that never deletes a pvc unless you explicitly do so) but https://github.com/elastic/cloud-on-k8s/issues/2328 has some explanations and solutions.

Basically we need to set a flag like this following in the elasticsearch crd to prevent the operator delete the pvc when the elasticsearch crd is deleted, which is implemented here https://github.com/elastic/cloud-on-k8s/pull/4050.

```
apiVersion: elasticsearch.k8s.elastic.co/v1
kind: Elasticsearch
metadata:
  name: es
spec:
  version: 7.10.1
  volumeClaimDeletePolicy: DeleteOnScaledownAndClusterDeletion
  nodeSets:
  - name: default
    count: 2
```
